grails-app/services/org/pih/warehouse/receiving/ReceiptService.groovy

```diff
@@ -148,6 +148,13 @@ class ReceiptService {
         partialReceiptItem.lotNumber = shipmentItem.inventoryItem?.lotNumber
         partialReceiptItem.expirationDate = shipmentItem.inventoryItem?.expirationDate
         partialReceiptItem.quantityShipped = shipmentItem?.quantity ?: 0
+        if (!shipmentItem.inventoryItem) {
+            shipmentItem.inventoryItem =
+                    inventoryService.findOrCreateInventoryItem(
```

Comment:
This should be done in the StockTransferApiController and StockTransferService side. Debug why it is not working there.